lib/ace/ext/papeeria/papeeria_latex_highlight_rules.coffee

```
@@ -44,10 +44,16 @@ define((require, exports, module) ->
         return to
 
     popState = (currentState, stack) ->
-      top = stack.pop()
+      if not stack?
+        return 'start'
 
-      if stack?
+      stack.pop()
+      if stack.length == 0
         return 'start'
+
+      top = stack.pop()
```

So you silently skip the topmost element and return last but one. WHY?